### PR TITLE
Feature/how claims work

### PR DESCRIPTION
**Pull request by MathiasOhrstrom**
Item from design: [Home Tab #Start]

### Screenshots 
image
image
image
image
image

### Checklist

- [x] Functionality is covered by an integration test
- [x] Functionality is accessible in engineering mode

### CODE REVIEW

File: app/src/main/java/com/hedvig/app/feature/dismissiblepager/DismissiblePager.kt
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/dismissiblepager/DismissiblePager.kt:118]: I think we can perhaps remove these position checks in our new paradigm with the `DismissiblePagerModel`. Let's investigate that together.

File: app/src/main/java/com/hedvig/app/feature/dismissiblepager/DismissiblePagerAdapter.kt
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/dismissiblepager/DismissiblePagerAdapter.kt:11]: I think this constant might be deprecated, let's check that out

File: app/src/main/java/com/hedvig/app/feature/home/ui/HomeModel.kt
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/home/ui/HomeModel.kt:38]:
```suggestion
    data class HowClaimsWork(val pages: List<HomeQuery.HowClaimsWork>): HomeModel()
```

File: app/src/main/java/com/hedvig/app/feature/loggedin/ui/LoggedInActivity.kt
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/loggedin/ui/LoggedInActivity.kt:22]: Are these imports used?

File: app/src/main/java/com/hedvig/app/util/extensions/ContextExtensions.kt
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/util/extensions/ContextExtensions.kt:45]: I think we already have an extension function called `colorAttr` 🙂

**Review by hedvigoscar — CHANGES_REQUESTED** [review]
As there are a lot of changes to be made, let's pair up on fixing them 🙂

File: app/src/main/res/layout/how_claims_work_button.xml
Comment (MathiasOhrstrom) [reply app/src/main/res/layout/how_claims_work_button.xml:3]: I used the frameLayout to center the button

Comment (hedvigoscar) [reply app/src/main/res/layout/how_claims_work_button.xml:3]: Of course! That makes sense. We can disregard this.

File: app/src/main/java/com/hedvig/app/feature/dismissiblepager/DismissiblePageFragment.kt
Comment (MathiasOhrstrom) [reply app/src/main/java/com/hedvig/app/feature/dismissiblepager/DismissiblePageFragment.kt:15]: What do you mean?

File: app/src/main/java/com/hedvig/app/feature/dismissiblepager/DismissiblePager.kt
Comment (MathiasOhrstrom) [reply app/src/main/java/com/hedvig/app/feature/dismissiblepager/DismissiblePager.kt:96]: Do you mean from content-service?

File: app/src/main/java/com/hedvig/app/feature/dismissiblepager/DismissiblePager.kt
Comment (MathiasOhrstrom) [reply app/src/main/java/com/hedvig/app/feature/dismissiblepager/DismissiblePager.kt:118]: Dont we need these to check if we are on the page next to the `SwipeOffScreen` fragment?

File: app/src/main/java/com/hedvig/app/feature/dismissiblepager/DismissiblePageFragment.kt
Comment (hedvigoscar) [reply app/src/main/java/com/hedvig/app/feature/dismissiblepager/DismissiblePageFragment.kt:15]:
```kotlin
import androidx.fragment.app.Fragment
// ...

class DismissiblePagerFragment : Fragment() {
```

Comment (MathiasOhrstrom) [reply app/src/main/java/com/hedvig/app/feature/dismissiblepager/DismissiblePageFragment.kt:15]: Aahh 🤦

**Comment by hedvigoscar** [comment]
I really like the fixes to the bottom sheets and the navigation bar, but we should try to keep those changes in separate PRs in the future as the scope of this PR is starting to grow quite large 😬

**Comment by MathiasOhrstrom** [comment]
> I really like the fixes to the bottom sheets and the navigation bar, but we should try to keep those changes in separate PRs in the future as the scope of this PR is starting to grow quite large 😬

True i should probably have done that in i another PR, however here are the screenshots 
image
image

**Comment by hedvigoscar** [comment]
Looks amazing 🙂

File: app/src/main/java/com/hedvig/app/feature/home/ui/HomeAdapter.kt
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/home/ui/HomeAdapter.kt:334]: Move this if to the row where we decide which button text we send to the `NoTitlePage`

File: app/src/main/java/com/hedvig/app/feature/home/ui/HomeAdapter.kt
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/home/ui/HomeAdapter.kt:350]:
```suggestion
                        .show(fragmentManager, HowClaimsWorkDialog.TAG)
```

File: app/src/main/java/com/hedvig/app/feature/whatsnew/WhatsNewDialog.kt
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/whatsnew/WhatsNewDialog.kt:51]: Same here!

File: app/src/main/java/com/hedvig/app/feature/welcome/WelcomeViewModel.kt
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/welcome/WelcomeViewModel.kt:17]: There's some risk with doing this due to how we work with `Context` and language resolution. I've got a nice strategy for refactoring it, let's do it together right away.

**Review by hedvigoscar — COMMENTED** [review]
Overall very good stuff. Let's pair program the last parts so that we get it over the finish line!

**Review by hedvigoscar — COMMENTED** [review]
Approved pending design QA. Do not merge until that QA is done.

File: app/src/main/res/layout/dismissible_page_no_title_fragment.xml
Comment (hedvigoscar) [inline app/src/main/res/layout/dismissible_page_no_title_fragment.xml:15]: extract to dimens

Comment (hedvigoscar) [reply app/src/main/res/layout/dismissible_page_no_title_fragment.xml:15]: actually come to think of it, is this necessary? given that we have a fixed width/height.

**Review by hedvigoscar — APPROVED** [review]
Approved after checking comments

File: app/src/main/res/layout/dismissible_page_no_title_fragment.xml
Comment (MathiasOhrstrom) [reply app/src/main/res/layout/dismissible_page_no_title_fragment.xml]: Ture! I forgot to remove it

**Pull request merged**
